File: tf1testech.client/src/models/LeavePeriodCreationDTO.ts
import { LeavePeriodType } from "./enums/LeavePeriodType";

class LeavePeriodCreationDTO {
    employeeId: string;
    comment: string;
    private startDate: string;
    private endDate : string;
    type: LeavePeriodType;

    constructor(employeeId: string, comment: string, startDate: string, endDate: string, type: LeavePeriodType){
        this.employeeId = employeeId;
        this.comment = comment;
        this.startDate = startDate;
        this.endDate = endDate;
        this.type = type;
    }
}

export default LeavePeriodCreationDTO;

File: Domain/ValueObjects/LeavePeriodCreationDTO.cs
using Domain.Agregates;
using Domain.Entities;
using Domain.Entities.Enums;

namespace Domain.ValueObjects
{
    public record LeavePeriodCreationDTO(
        string EmployeeId,
        string Comment,
        DateTime StartDate,
        DateTime EndDate,
        LeavePeriodType Type)
    {
        public LeavePeriod ToDomain()
        {
            return new LeavePeriod()
            {
                Employee = new Employee(EmployeeId),
                LeavePeriodDemand = new LeavePeriodDemand()
                {
                    Comment = Comment,
                    StartDate = StartDate,
                    EndDate = EndDate,
                    Type = Type
                }
            };
        }
    }
}
